src/llm/runtime.py

```
@@ -25,39 +27,215 @@
+        logger.debug("Failed to open Langfuse agent step: %s", exc)
+        yield
+        return
+
+    with obs_cm:
+        yield
+
+
+def annotate_langfuse_run_io(
+    run_span: Any,
+    *,
+    input: Any = None,
```

Ruff (0.15.17)

[error] Function argument `input` is shadowing a Python builtin

(A002)